### PR TITLE
Closes #2400 Unified input date object for `derive_vars_dt()` test suite

### LINKED ISSUE #2400: Create united example for test suite of `derive_vars_dt()`
The input object for `derive_vars_dt()` test suite changes throughout the script, we should run just one that has:

- complete date
- missing year
- missing month & day
- missing day
- missing only month
- empty string
- NA_character_
- inappropiate string format

### PR DESCRIPTION
**Pull request by zdz2101**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Review the [Cheat Sheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf). Make any required updates to it by editing the file `inst/cheatsheet/admiral_cheatsheet.pptx` and re-upload a PDF version of it to the same folder.
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: tests/testthat/test-derive_vars_dt.R [hunk 1]
```diff
@@ -1,105 +1,140 @@
-# impute_dtc_dt ----
 input <- c(
-  "2019-07-18",
-  "2019-02",
-  "2019",
-  "2019---07"
+  "2019-07-18", # full date
```
Comment (zdz2101) [inline tests/testthat/test-derive_vars_dt.R:2]: @bms63 @millerg23 @StefanThoma before I proceed further down the function/more functions, this is roughly the thing I had in mind, but then in occurred to me that we need to suppress warnings a lot, any thoughts?

Comment (zdz2101) [reply tests/testthat/test-derive_vars_dt.R:2]: Well it's more for the date-ish looking ones we expected a matched output, but where we expect things to go wrong, we want the warnings, but the value should map over to `NA`

Comment (bms63) [reply tests/testthat/test-derive_vars_dt.R:2]: yeah - maybe a bad data/extreme example would be nice to test things separately?

Comment (zdz2101) [reply tests/testthat/test-derive_vars_dt.R:2]: Makes more sense that way, will do

File: tests/testthat/test-derive_vars_dt.R [hunk 2]
```diff
@@ -1,105 +1,140 @@
...
 ## Test 4: impute month and day to last and preserve = TRUE ----
 test_that("impute_dtc_dt Test 4: impute month and day to last and preserve = TRUE", {
   expected_output <- c(
     "2019-07-18",
-    "2019-02-28",
+    NA_character_,
     "2019-12-31",
-    "2019-12-07"
+    "2019-07-31",
+    "2019-12-07",
```
Comment (StefanThoma) [inline tests/testthat/test-derive_vars_dt.R:97]: To me, it is strange that here only the month is imputed to last where in impute_dtc_dt Test 2 both the month and the day is imputed to first. 
This does not seem consistent to me.  @zdz2101 what is the rational behind this?

**Comment by StefanThoma** [comment]
I had a look and it looks really good!
I think the suppression of the warnings is fine in this context.
The warnings themselves should be checked in a separate test IMO.

**Comment by zdz2101** [comment]
@bms63 @StefanThoma @millerg23 Added in a separate input object that plugs in for its appropriate warnings but also checks the return object is still valid

**Review by bms63 — APPROVED** [review]

**Pull request merged**